Invalid type definition for mpwriter.append

### Describe the bug

It would appear that the `append` method of the MultiPartWriter has an incorrect type definition:

https://github.com/aio-libs/aiohttp/blob/d8936aacf7d602a58228d95fd469bbf01a43216d/aiohttp/multipart.py#L856

Resulting in type checking tools to incorrectly errpr what (I believe) should be correct.

### To Reproduce

The following test file generates the error in type checkers.

```py
import aiohttp

with aiohttp.MultipartWriter() as mpwriter:
    mpwriter.append("Hello, World", {"Content-Type": "text/plain"})
```

which generates the following errors:

```console
❯ mypy test.py
test.py:4: error: Argument 2 to "append" of "MultipartWriter" has incompatible type "dict[str, str]"; expected "MultiMapping[str] | None"  [arg-type]
Found 1 error in 1 file (checked 1 source file)

❯ pyright test.py
/Users/joshua.ellis/src/pact-foundation/pact-python/tests/v3/test.py
  /Users/joshua.ellis/src/pact-foundation/pact-python/tests/v3/test.py:4:37 - error: Argument of type "dict[str, str]" cannot be assigned to parameter "headers" of type "MultiMapping[str] | None" in function "append"
    Type "dict[str, str]" cannot be assigned to type "MultiMapping[str] | None"
      "dict[str, str]" is incompatible with "MultiMapping[str]"
      Type cannot be assigned to type "None" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

### Expected behavior

I expected that a regular Python dictionary be compatible with the header argument.

### Logs/tracebacks

```python-traceback
_see above_
```


### Python Version

```console
❯ python --version
Python 3.11.6
```


### aiohttp Version

```console
❯ python -m pip show aiohttp
Name: aiohttp
Version: 3.8.6
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by:
```


### multidict Version

```console
❯ python -m pip show multidict
Name: multidict
Version: 6.0.4
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Requires: 
Required-by: aiohttp, yarl
```


### yarl Version

```console
❯ python -m pip show yarl
Name: yarl
Version: 1.9.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Requires: idna, multidict
Required-by: aiohttp
```


### OS

```console
❯ uname -mprs
Darwin 22.6.0 arm64 arm
```

### Related component

Client

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

--- a/CHANGES/7741.bugfix.rst
+++ b/CHANGES/7741.bugfix.rst
@@ -1,7 +1,3 @@
-Fixed the type annotations on :meth:`MultipartWriter.append`,
+Changed the type annotations to allow dicts on :meth:`MultipartWriter.append`,
 :meth:`MultipartWriter.append_json` and
 :meth:`MultipartWriter.append_form` -- by :user:`cakemanny`
-
-Type checkers, such as mypy and Pyright, now no longer give type
-errors when calling :py:meth:`MultipartWriter.append` with a plain
-python dictionary as the headers argument.

--- a/CHANGES/7741.bugfix.rst
+++ b/CHANGES/7741.bugfix.rst
@@ -1,3 +1,3 @@
-Changed the type annotations to allow ``dict`` on :meth:`MultipartWriter.append`,
-:meth:`MultipartWriter.append_json` and
-:meth:`MultipartWriter.append_form` -- by :user:`cakemanny`
+Changed the type annotations to allow ``dict`` on :meth:`aiohttp.MultipartWriter.append`,
+:meth:`aiohttp.MultipartWriter.append_json` and
+:meth:`aiohttp.MultipartWriter.append_form` -- by :user:`cakemanny`

--- a/CHANGES/7741.bugfix.rst
+++ b/CHANGES/7741.bugfix.rst
@@ -1,3 +1,3 @@
-Changed the type annotations to allow dicts on :meth:`MultipartWriter.append`,
+Changed the type annotations to allow ``dict`` on :meth:`MultipartWriter.append`,
 :meth:`MultipartWriter.append_json` and
 :meth:`MultipartWriter.append_form` -- by :user:`cakemanny`

--- a/CHANGES/7741.bugfix.rst
+++ b/CHANGES/7741.bugfix.rst
@@ -1,7 +1,7 @@
-Fixed the type annotations on :py:meth:`MultipartWriter.append`,
-:py:meth:`MultipartWriter.append_json` and
-:py:meth:`MultipartWriter.append_form` -- by :user:`cakemanny`
+Fixed the type annotations on :meth:`MultipartWriter.append`,
+:meth:`MultipartWriter.append_json` and
+:meth:`MultipartWriter.append_form` -- by :user:`cakemanny`
 
-Typecheckers, such as mypy and pyright, now no longer give type
+Type checkers, such as mypy and Pyright, now no longer give type
 errors when calling :py:meth:`MultipartWriter.append` with a plain
 python dictionary as the headers argument.

--- a/CHANGES/7741.bugfix.rst
+++ b/CHANGES/7741.bugfix.rst
@@ -1,0 +1,7 @@
+Fixed the type annotations on :py:meth:`MultipartWriter.append`,
+:py:meth:`MultipartWriter.append_json` and
+:py:meth:`MultipartWriter.append_form` -- by :user:`cakemanny`
+
+Typecheckers, such as mypy and pyright, now no longer give type
+errors when calling :py:meth:`MultipartWriter.append` with a plain
+python dictionary as the headers argument.

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -83,6 +83,7 @@ Damien Nadé
 Dan King
 Dan Xu
 Daniel García
+Daniel Golding
 Daniel Grossmann-Kavanagh
 Daniel Nelson
 Daniele Trifirò

--- a/aiohttp/multipart.py
+++ b/aiohttp/multipart.py
@@ -25,7 +25,8 @@ from typing import (
 )
 from urllib.parse import parse_qsl, unquote, urlencode
 
-from multidict import CIMultiDict, CIMultiDictProxy, MultiMapping
+from multidict import CIMultiDict, CIMultiDictProxy
+
 
 from .compression_utils import ZLibCompressor, ZLibDecompressor
 from .hdrs import (
@@ -47,6 +48,7 @@ from .payload import (
     payload_type,
 )
 from .streams import StreamReader
+from .typedefs import LooseHeaders
 
 __all__ = (
     "MultipartReader",
@@ -853,7 +855,7 @@ class MultipartWriter(Payload):
     def boundary(self) -> str:
         return self._boundary.decode("ascii")
 
-    def append(self, obj: Any, headers: Optional[MultiMapping[str]] = None) -> Payload:
+    def append(self, obj: Any, headers: Optional[LooseHeaders] = None) -> Payload:
         if headers is None:
             headers = CIMultiDict()
 
@@ -901,7 +903,7 @@ class MultipartWriter(Payload):
         return payload
 
     def append_json(
-        self, obj: Any, headers: Optional[MultiMapping[str]] = None
+        self, obj: Any, headers: Optional[LooseHeaders] = None
     ) -> Payload:
         """Helper to append JSON part."""
         if headers is None:
@@ -912,7 +914,7 @@ class MultipartWriter(Payload):
     def append_form(
         self,
         obj: Union[Sequence[Tuple[str, str]], Mapping[str, str]],
-        headers: Optional[MultiMapping[str]] = None,
+        headers: Optional[LooseHeaders] = None,
     ) -> Payload:
         """Helper to append form urlencoded part."""
         assert isinstance(obj, (Sequence, Mapping))

--- a/aiohttp/multipart.py
+++ b/aiohttp/multipart.py
@@ -47,7 +47,6 @@ from .payload import (
     payload_type,
 )
 from .streams import StreamReader
-from .typedefs import LooseHeaders
 
 __all__ = (
     "MultipartReader",
@@ -854,7 +853,7 @@ class MultipartWriter(Payload):
     def boundary(self) -> str:
         return self._boundary.decode("ascii")
 
-    def append(self, obj: Any, headers: Optional[LooseHeaders] = None) -> Payload:
+    def append(self, obj: Any, headers: Optional[Mapping[str, str]] = None) -> Payload:
         if headers is None:
             headers = CIMultiDict()
 
@@ -901,7 +900,7 @@ class MultipartWriter(Payload):
         self._parts.append((payload, encoding, te_encoding))  # type: ignore[arg-type]
         return payload
 
-    def append_json(self, obj: Any, headers: Optional[LooseHeaders] = None) -> Payload:
+    def append_json(self, obj: Any, headers: Optional[Mapping[str, str]] = None) -> Payload:
         """Helper to append JSON part."""
         if headers is None:
             headers = CIMultiDict()
@@ -911,7 +910,7 @@ class MultipartWriter(Payload):
     def append_form(
         self,
         obj: Union[Sequence[Tuple[str, str]], Mapping[str, str]],
-        headers: Optional[LooseHeaders] = None,
+        headers: Optional[Mapping[str, str]] = None,
     ) -> Payload:
         """Helper to append form urlencoded part."""
         assert isinstance(obj, (Sequence, Mapping))

--- a/aiohttp/multipart.py
+++ b/aiohttp/multipart.py
@@ -27,7 +27,6 @@ from urllib.parse import parse_qsl, unquote, urlencode
 
 from multidict import CIMultiDict, CIMultiDictProxy
 
-
 from .compression_utils import ZLibCompressor, ZLibDecompressor
 from .hdrs import (
     CONTENT_DISPOSITION,
@@ -902,9 +901,7 @@ class MultipartWriter(Payload):
         self._parts.append((payload, encoding, te_encoding))  # type: ignore[arg-type]
         return payload
 
-    def append_json(
-        self, obj: Any, headers: Optional[LooseHeaders] = None
-    ) -> Payload:
+    def append_json(self, obj: Any, headers: Optional[LooseHeaders] = None) -> Payload:
         """Helper to append JSON part."""
         if headers is None:
             headers = CIMultiDict()

--- a/aiohttp/multipart.py
+++ b/aiohttp/multipart.py
@@ -900,7 +900,9 @@ class MultipartWriter(Payload):
         self._parts.append((payload, encoding, te_encoding))  # type: ignore[arg-type]
         return payload
 
-    def append_json(self, obj: Any, headers: Optional[Mapping[str, str]] = None) -> Payload:
+    def append_json(
+        self, obj: Any, headers: Optional[Mapping[str, str]] = None
+    ) -> Payload:
         """Helper to append JSON part."""
         if headers is None:
             headers = CIMultiDict()

